### DISCUSSION TITLE
Issues with Email Sending in HotCRP Configuration

### DISCUSSION THREAD

**Discussion by Alizadeh-ACoM** [kohler/hotcrp#372] [General]
Hello HotCRP Team,

I am encountering issues with email sending functionality in HotCRP. Here’s the situation and steps I’ve taken to debug the problem:
Environment Details
HotCRP Version: (Please include your version)
Server: Ubuntu (or your OS)
PHP: 8.2 (via php8.2-fpm)
SMTP Server: mail.rwth-aachen.de

Here’s my configuration in options.php:

$Opt["dbName"] = "hotcrp";
$Opt["dbUser"] = "**";
$Opt["dbPassword"] = "**";
$Opt["shortName"] = "My Conference";
$Opt["longName"] = "Conference Management System";
$Opt["contactEmail"] = "a**e*;

// SMTP server configuration
$Opt["smtpServer"] = "mail.rwth-aachen.de";
$Opt["smtpPort"] = 587;
$Opt["smtpUsername"] = "**";
$Opt["smtpPassword"] = "**";
$Opt["smtpSecure"] = "tls";
$Opt["emailFrom"] = "a**e";
$Opt["emailSender"] = "a**e";
$Opt["sendEmail"] = true;

// Debugging enabled
$Opt["debug"] = true;
$Opt["mailErrorCallback"] = function ($email, $error) {
    error_log("Failed to send email to {$email['to']}: {$error}");
};

I’ve successfully sent test emails using the following PHPMailer code, This confirms that my SMTP settings work as expected.
But when creating accounts or sending password reset emails via HotCRP, the interface displays a success message (e.g., "Sent mail to ..."), but no emails are received.

Logs:
action.log: No entries appear when emails are triggered.
/var/log/mail.log: No entries are generated by HotCRP actions.

Questions
Are there specific debug steps to trace why HotCRP’s internal mail system isn’t working despite correct SMTP settings?
Could this be an issue with email templates, or does HotCRP’s mailer.php need additional configuration for TLS?
Is there a specific log file or error handling configuration that I missed?
Thank you for your assistance!

**Comment by kohler** [comment]
What “following PHPMailer code”?

Where did you get the idea to set options like `$Opt["smtpPort"]` etc.? $Opt is for HotCRP options, and HotCRP does not support any of those `smtp` options.

I don't generally provide support for debugging mail configurations because mail is super shitty to set up.
